packages/zoom-in-ui/src/components/app.js

```diff
@@ -0,0 +1,230 @@
+    return searchParams.get(query)
+  }
+}
+
+const adoptTwreporterTheme = (image, caption) => image && caption
+
+const App = props => {
+  const { description, title } = props
+  const classes = useStyles()
+  const [imageLink, setImageLink] = useState(
+    getParamsFromSearch(searchKey.image) || ''
+  )
+  const [imageCaption, setImageCaption] = useState(
+    getParamsFromSearch(searchKey.caption) || ''
+  )
+  const [code, setCode] = useState(null)
+  const [buildCodeError, setBuildCodeError] = useState(null)
+  const [theme, setTheme] = useState(
+    _.merge(
+      {},
+      adoptTwreporterTheme(imageLink, imageCaption)
+        ? themes.twreporterTheme
+        : themes.defaultTheme
+    )
+  )
+
+  const handleSaveLink = ({ imgUrl, caption }) => {
+    const trimmedLink = imgUrl.trim()
+    if (trimmedLink.length > 0) {
+      const newTheme = adoptTwreporterTheme(trimmedLink, caption)
+        ? themes.twreporterTheme
+        : themes.defaultTheme
+      setImageLink(trimmedLink)
+      setImageCaption(caption)
+      setTheme(_.merge({}, newTheme))
+    }
+  }
+
+  const updateTheme = updatedPart => {
+    const updatedTheme = _.merge({}, theme, updatedPart)
+    setTheme(updatedTheme)
+  }
+
+  const buildCode = () => {
+    try {
+      const code = zoomIn.buildEmbeddedCode(
+        {
+          data: {
+            src: imageLink,
+            alt: imageCaption,
+            caption: imageCaption,
+            theme,
+          },
+        },
+        webpackAssets
+      )
+      setCode(code)
+      setBuildCodeError(null)
+    } catch (error) {
+      setBuildCodeError(error)
+    }
+  }
+
+  const ZoomableImage = zoomIn.Component
+  return (
+    <>
+      <CssBaseline />
+      <Container maxWidth="sm">
+        <Paper className={classes.paper}>
+          <Typography variant="h3" component="h1" gutterBottom>
+            {title}
+          </Typography>
+          <Typography variant="body1" component="div" gutterBottom>
+            {_.map(description, (p, i) => (
+              <p key={i} dangerouslySetInnerHTML={{ __html: p }} />
+            ))}
+          </Typography>
+          <Divider className={classes.divider} variant="middle" />
+          <AddImageForm
+            buttonLabel="Submit"
+            placeholder={[
+              'public URL of your image',
+              'caption for the image (optional)',
+            ]}
+            submitHandler={handleSaveLink}
+            icon={<ArrowDownwardIcon />}
+            defaultValue={{
+              image: imageLink,
+              caption: imageCaption,
+            }}
+          />
+          {imageLink ? (
+            <div className={classes.sandbox}>
+              <Setting
+                theme={theme}
+                updateTheme={updateTheme}
+                caption={imageCaption}
+              />
+              <ZoomableImage
+                src={imageLink}
+                caption={imageCaption}
+                theme={theme}
```

Comment:
User experience advice:
I test `zoom-in-ui` locally, and I paste the url with `https://www.twreporter.org/images/20160602104752-0c77d34f55ccafd945b9f1e55438348b.jpg`.
Due to cors setting, that image doesn't support cross origin sharing.
Hence, we cannot see any results in the preview section.
At first sight, I thought our service is broken. 
However, the service works at all.
Maybe we could show some messages if the image is not fully reachable?

<img width="481" alt="Screen Shot 2020-10-12 at 4 41 01 PM" src="https://user-images.githubusercontent.com/3000343/95725111-cc358d00-0ca9-11eb-908e-3caac68fe85b.png">

Reply:
Updated.

Please see commit [1d06323](https://github.com/twreporter/orangutan-monorepo/pull/70/commits/1d06323c11d6dc06c0c17efdf497bbf596bd2ad6)